client/gutenberg/extensions/email-subscribe/index.js

```diff
@@ -0,0 +1,37 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import EmailSubscribeEdit from './edit';
+import './editor.scss';
+
+export const name = 'email-subscribe';
+
+export const settings = {
+	title: __( 'Email Subscribe', 'jetpack' ),
+	icon: 'email',
+	category: 'widgets',
+	keywords: [ __( 'email', 'jetpack' ), __( 'mailchimp', 'jetpack' ), 'jetpack' ],
+
+	edit: EmailSubscribeEdit,
+
+	// We will generate shortcode as a fallback, so that this works even if gutenberg does not.
+	save: function( props ) {
+		let attrs = '';
+		if ( props.attributes ) {
+			attrs =
+				' ' +
+				Object.keys( props.attributes )
+					.filter( function( key ) {
+						return props.attributes[ key ];
+					} )
+					.map( function( key ) {
+						return key + '="' + props.attributes[ key ].replace( '"', "'" ) + '"';
+					} )
+					.join( ' ' );
+		}
+		return '[jetpack-email-subscribe' + attrs + ']';
```

Comment:
If I understood correctly what happens with `attrs`, I think you can use Gutenberg's shortcode instance directly:

```js
import shortcode from '@wordpress/shortcode';
new shortcode({ tag: 'jetpack-email-subscribe', attrs: props.attributes }).string();
```

See more instance options from:
https://github.com/WordPress/gutenberg/blob/64045784bac1b29a243ba8beae33c31e887c2fe3/packages/shortcode/src/index.js

Reply:
You can test the above in Gutenberg editor window via console:

```
new wp.shortcode({ tag: 'demo' }).string();
```

Reply:
That is neat!
I opted in to remove whole shortcode building since that was a stopgap in case 5.0 does not make it in long time, but its out, so no need for that.